### PR TITLE
Send file system manifest up on want_config

### PR DESCRIPTION
**Pull request by thebentern**

### CODE REVIEW

File: src/mesh/PhoneAPI.cpp [hunk 1]
```diff
@@ -340,13 +345,28 @@ size_t PhoneAPI::getFromRadio(uint8_t *buf)
...
             // Go ahead and send that ID right now
             return getFromRadio(buf);
         }
         break;
     }
 
+    case STATE_SEND_FILEMANIFEST: {
+        LOG_INFO("getFromRadio=STATE_SEND_FILEMANIFEST\n");
+        fromRadioScratch.which_payload_variant = meshtastic_FromRadio_fileInfo_tag;
+        if (config_state < filesManifest.size()) {
```
Comment (mverch67) [inline src/mesh/PhoneAPI.cpp:358]: Has this ever been tested? What if size() returns 0? config_state is **unsigned**, can never be < 0

File: src/mesh/PhoneAPI.cpp [hunk 2]
```diff
@@ -340,13 +345,28 @@ size_t PhoneAPI::getFromRadio(uint8_t *buf)
...
+            fromRadioScratch.fileInfo = filesManifest.at(config_state);
+            config_state++;
+            // last element
+            if (config_state == filesManifest.size()) {
+                state = STATE_SEND_COMPLETE_ID;
```
Comment (mverch67) [inline src/mesh/PhoneAPI.cpp:363]: This line will never be reached ==> endless loop

**Comment by mverch67** [comment]
What is this `SPECIAL_NONCE 69420`  used for? There is no comment in the code !?

File: src/mesh/PhoneAPI.cpp [hunk 3]
```diff
@@ -324,8 +328,9 @@ size_t PhoneAPI::getFromRadio(uint8_t *buf)
         // Advance when we have sent all of our ModuleConfig objects
         if (config_state > (_meshtastic_AdminMessage_ModuleConfigType_MAX + 1)) {
             // Clients sending special nonce don't want to see other nodeinfos
-            state = config_nonce == SPECIAL_NONCE ? STATE_SEND_COMPLETE_ID : STATE_SEND_OTHER_NODEINFOS;
+            state = config_nonce == SPECIAL_NONCE ? STATE_SEND_FILEMANIFEST : STATE_SEND_OTHER_NODEINFOS;
             config_state = 0;
+            filesManifest.clear();
```
Comment (mverch67) [inline src/mesh/PhoneAPI.cpp:333]: Looks like this vector is emptied too early, so size() will be 0 when the FILEMANIFEST is supposed to be sent.

File: src/mesh/PhoneAPI.cpp (hunk 1 above)
Comment (thebentern) [reply src/mesh/PhoneAPI.cpp:358]: Received: { "fileInfo": { "fileName": "/prefs/channels.proto", "sizeBytes": 57 } }
Received: { "fileInfo": { "fileName": "/prefs/config.proto", "sizeBytes": 98 } }
Received: { "fileInfo": { "fileName": "/prefs/db.proto", "sizeBytes": 700 } }
Received: { "fileInfo": { "fileName": "/prefs/module.proto", "sizeBytes": 104 } }
Received: { "fileInfo": { "fileName": "/static/.gitkeep" } }
Received: { "fileInfo": { "fileName": "/static/Logo_Black.svg.gz", "sizeBytes": 602 } }
Received: { "fileInfo": { "fileName": "/static/Logo_White.svg.gz", "sizeBytes": 610 } }
Received: { "fileInfo": { "fileName": "/static/apple-touch-icon.png.gz", "sizeBytes": 3164 } }
Received: { "fileInfo": { "fileName": "/static/favicon.ico.gz", "sizeBytes": 2270 } }
Received: { "fileInfo": { "fileName": "/static/icon.svg.gz", "sizeBytes": 852 } }
Received: { "fileInfo": { "fileName": "/static/index-B272zAn3.css.gz", "sizeBytes": 16427 } }
Received: { "fileInfo": { "fileName": "/static/index-CGqDWSD_.js.gz", "sizeBytes": 348 } }
Received: { "fileInfo": { "fileName": "/static/index-CgmZQ_N-.js.gz", "sizeBytes": 204401 } }
Received: { "fileInfo": { "fileName": "/static/index.html.gz", "sizeBytes": 549 } }
Received: { "fileInfo": { "fileName": "/static/maplibre-gl-69L4OEQ1.js.gz", "sizeBytes": 210430 } }
Received: { "fileInfo": { "fileName": "/static/robots.txt.gz", "sizeBytes": 42 } }
Received: { "fileInfo": { "fileName": "/static/site.webmanifest.gz", "sizeBytes": 197 } }
Received: { "configCompleteId": 1567494388 }

**Pull request merged**